tiledbcontents/call_server.py

```diff
@@ -0,0 +1,100 @@
+"""Extremely simple server process which makes calls and returns results.
+
+We use this to separate out calls which use TileDB core libraries from
+the process which runs the rest of JupyterLab.
+"""
+
+import argparse
+import os
+import socketserver
+import threading
+import time
+from typing import Generic, NoReturn, TypeVar
+
+import attrs
+import cloudpickle
+
+_T_co = TypeVar("_T_co", covariant=True)
+
+
+@attrs.define(frozen=True, slots=True)
+class Success(Generic[_T_co]):
+    """Wrapper for a successful return."""
+
+    value: _T_co
+
+    def result(self) -> _T_co:
+        return self.value
+
+
+@attrs.define(frozen=True, slots=True)
+class Failure:
+    """Wrapper for a failed return."""
+
+    exception: Exception
+
+    def result(self) -> NoReturn:
+        raise self.exception
+
+
+class Executor(socketserver.StreamRequestHandler):
+    def handle(self) -> None:
+        result: object
+        try:
+            req = cloudpickle.load(self.rfile)
+        except Exception as e:
+            result = Failure(e)
+        else:
+            if req is None:
+                # This is the last request we should handle.
+                self.server.shutdown()
+                return
+            try:
+                result = Success(req())
+            except Exception as e:
+                result = Failure(e)
+        cloudpickle.dump(result, self.wfile)
```

Comment:
Should we put a guard on the return types here? If we return a pickleable tiledb-py object, I believe we may implicitly cause `import tiledb` in the client process.

Reply:
Hmm, ok - it's a bit fraught though because for Arrays (for example) we just serialize the URI and timestamp, and reopen. So if someone accidentally returned an Array object, I think it would implicitly cause a context to be created.